src/controllers/mainController.ts

```diff
@@ -2598,12 +2610,38 @@ export default class MainController implements vscode.Disposable {
         return await this.newQueryFromPrompt(newDocUri);
     }
 
-    public async onSchemaCompare(node?: any): Promise<void> {
+    /**
+     * Handler for the Schema Compare command.
+     * Accepts variable arguments, typically:
+     *   - [sourceNode, targetNode, runComparison] when invoked from update Project SC or programmatically,
+     *   - [sourceNode, undefined] when invoked from a project tree node/ server / database node,
+     *   - [] when invoked from the command palette.
+     * This method normalizes the arguments and launches the Schema Compare UI.
+     */
+    public async onSchemaCompare(
+        ...args: (ConnectionNode | TreeNodeInfo | SchemaCompareEndpointInfo | boolean | undefined)[]
+    ): Promise<void> {
+        let sourceNode = undefined;
```

Comment:
This arg parsing should be in the function above.  We want to marshall the args into known variables and types/unions as soon as possible.

Reply:
This can happen in a later PR